optaplanner-core/src/test/java/org/optaplanner/core/config/solver/SolverConfigTest.java

```diff
@@ -19,66 +19,43 @@
 
-import javax.xml.bind.JAXBContext;
-import javax.xml.bind.JAXBException;
-import javax.xml.bind.Marshaller;
-import javax.xml.bind.Unmarshaller;
-import javax.xml.transform.OutputKeys;
-import javax.xml.transform.Transformer;
-import javax.xml.transform.TransformerException;
-import javax.xml.transform.TransformerFactory;
-import javax.xml.transform.dom.DOMResult;
-import javax.xml.transform.dom.DOMSource;
-import javax.xml.transform.stream.StreamResult;
-
 import org.apache.commons.io.IOUtils;
 import org.junit.jupiter.api.Test;
 import org.optaplanner.core.api.score.stream.ConstraintProvider;
-import org.optaplanner.core.api.solver.SolverFactory;
 import org.optaplanner.core.config.constructionheuristic.ConstructionHeuristicPhaseConfig;
 import org.optaplanner.core.config.heuristic.selector.move.generic.ChangeMoveSelectorConfig;
 import org.optaplanner.core.config.heuristic.selector.value.ValueSelectorConfig;
 import org.optaplanner.core.impl.heuristic.selector.common.decorator.SelectionFilter;
 import org.optaplanner.core.impl.heuristic.selector.move.factory.MoveIteratorFactory;
 import org.optaplanner.core.impl.heuristic.selector.move.factory.MoveListFactory;
 import org.optaplanner.core.impl.heuristic.selector.move.generic.ChangeMove;
+import org.optaplanner.core.impl.io.XmlIO;
+import org.optaplanner.core.impl.io.jaxb.JaxbIO;
 import org.optaplanner.core.impl.partitionedsearch.partitioner.SolutionPartitioner;
 import org.optaplanner.core.impl.score.director.easy.EasyScoreCalculator;
 import org.optaplanner.core.impl.score.director.incremental.IncrementalScoreCalculator;
-import org.optaplanner.core.impl.solver.io.XStreamConfigReader;
 import org.optaplanner.core.impl.testdata.domain.TestdataEntity;
 import org.optaplanner.core.impl.testdata.domain.TestdataSolution;
 import org.optaplanner.core.impl.testdata.domain.TestdataValue;
 
-import com.thoughtworks.xstream.XStream;
-
 public class SolverConfigTest {
     private static final String TEST_SOLVER_CONFIG = "testSolverConfig.xml";
 
-    private final Unmarshaller unmarshaller;
-    private final Marshaller marshaller;
-
-    public SolverConfigTest() throws JAXBException {
-        JAXBContext jaxbContext = JAXBContext.newInstance(SolverConfig.class);
-        unmarshaller = jaxbContext.createUnmarshaller();
-        marshaller = jaxbContext.createMarshaller();
-        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
-        marshaller.setProperty(Marshaller.JAXB_ENCODING, StandardCharsets.UTF_8.toString());
-    }
+    private final XmlIO<SolverConfig> xmlIO = new JaxbIO<>(SolverConfig.class);
 
     @Test
     public void jaxbXmlConfigFileRemainsSameAfterReadWrite() throws IOException {
         SolverConfig jaxbSolverConfig = unmarshallSolverConfigFromResource(TEST_SOLVER_CONFIG);
 
         Writer stringWriter = new StringWriter();
-        marshall(jaxbSolverConfig, stringWriter);
+        xmlIO.write(jaxbSolverConfig, stringWriter);
```

Comment:
Wouldn't this warn of try-with-resources violation?
(It's not an issue, since `StringWriter` can't leak anything, but I don't think IDEs are so smart.)

Reply:
No warning by my IDE at all.